oio/directory/meta0.py

```diff
@@ -201,6 +199,62 @@ def __init__(self, meta0_client, replicas=3,
         if self.digits > 4:
             raise ConfigurationException("meta_digits must be <= 4")
         self.min_dist = min_dist
+        if self.min_dist < 1:
+            raise ConfigurationException("min_dist must be >= 1")
+        if self.min_dist > 4:
+            raise ConfigurationException("min_dist must be <= 4")
+        super(Meta0PrefixMapping, self).__init__(
+            meta0_client.conf, ['meta1'], **kwargs)
+
+    def reset(self):
+        super(Meta0PrefixMapping, self).reset()
+
+        for svc_id, svc in self.services.iteritems():
+            location = self.get_loc(svc)
+            if self.min_dist == 1:
+                site = location
+            else:
+                site = '.'.join(location.split('.')[:1-self.min_dist])
+            svc['site'] = site
+            svc['upper_limit'] = 0
+
+    def _compute_upper_limit(self):
+        available_svcs_by_site = dict()
+
+        for svc_id, svc in self.services.iteritems():
+            if self.get_score(svc) <= 0:
+                continue
+            site = svc['site']
+            available_svcs_by_site[site] = \
+                available_svcs_by_site.setdefault(site, 0) + 1
```

Comment:
Since the value is an integer, no need for `setdefault` here, `get` is sufficient.